Proofreading SAP2 (#6673)
mendix/docs: content/en/docs/developerportal/deploy/sap-btp/sap-autoscaler.md

## 1 Introduction

The application autoscaler service for [SAP Business Technology Platform (SAP BTP)](https://docs.mendix.com/developerportal/deploy/sap-cloud-platform/) enables applications to automatically increase or decrease the number of instances of your app. This service can be based on a schedule, or based on the value of a metric.

For more information on the application autoscaler, see [Application Autoscaler Service](https://help.sap.com/docs/CSI/b651ed9a533d41339f05b12549d8d706/c5d349a5ea354971839b7922d13c04e4.html?q=application%20autoscaler%20service) document on the SAP Help Portal.

## 2 Binding the Application Autoscaler{#binding}

The **autoscaler** service can be found on the [Services Tab](/developerportal/deploy/sap-cloud-platform/#binding-services) of the environment details page of your app's environment.

> Selecting the **autoscaler** service, expands the entry to allow you to enter more details about the service you wish to bind, namely the **Plan** and the **File** which contains the configuration.

Reviewer: Don't think we need a comma before "expands", it is not an introductory phrase.

Author: done https://github.com/mendix/docs/pull/6673/commits/be785809847f6b0b1d0a54e88bccaceb697d2a72

{{< figure src="/attachments/developerportal/deploy/sap-autoscaler/autoscaler-service.png" alt="Autoscaler service on the environment details page" >}}

You need to specify the following:

* **Plan** – the plan you wish to use for the autoscaler, either *Lite* or *Standard*

> * **File** – the file which contains the JSON to configure the autoscaler — more information, including the contents of the configuration file can be found in [Application Autoscaler Service](https://help.sap.com/docs/CSI/b651ed9a533d41339f05b12549d8d706/c5d349a5ea354971839b7922d13c04e4.html?q=application%20autoscaler%20service) on the SAP Help Portal

Reviewer: This new link does not contain any information about the JSON file. Can you check with the development team if this is a better suggestion?
https://help.sap.com/docs/application-autoscaler/application-autoscaler/defining-scaling-policy

Author: Verified this link with developer and corrected. https://github.com/mendix/docs/pull/6673/commits/61678ca354771c2dc2d3f72944b74c1494af416b

Having provided this information, you can click **Connect Services** to connect the autoscaler to your app.

The structure of the configuration file can get quite complex, as described in the documentation of SAP website. Mendix therefore provides you with a tool to help you create the correct configuration.

Click **Configurator** to start the **Autoscaler Configurator**.

## 3 Using the Autoscaler Configurator

The **Autoscaler Configurator** provides a user-friendly interface to create the JSON file required to configure the application autoscaler.

{{% alert color="info" %}}
The **Configurator** only creates a JSON file from scratch. It does not have access to your existing autoscaler settings on [SAP BTP](https://docs.mendix.com/developerportal/deploy/sap-cloud-platform/), and you cannot import an existing JSON files.
{{% /alert %}}

There are three options available:

* Scheduled Scaling – the app is scaled according to a fixed schedule
* Dynamic Scaling – the app is scaled depending on the value of a metric
* Custom – a combination of both scheduled and dynamic scaling

The metrics you can use to trigger dynamic scaling are as follows:

> * Memory consumed - monitors memory consumption in megabytes (memoryconsumed) or in percentage (memoryutil)

Reviewer: These hyphens should be n-dashes `–` not hyphens `-`
```suggestion
* Memory consumed – monitors memory consumption in megabytes (memoryconsumed) or in percentage (memoryutil)
```

Author: done https://github.com/mendix/docs/pull/6673/commits/be785809847f6b0b1d0a54e88bccaceb697d2a72

* CPU utilization in percentage
* Response time - monitors response time in milliseconds
* Throughput - monitors requests per seconds (RPS)

{{% alert color="info" %}}
The **Configurator** does not support adding custom metrics as triggers.
{{% /alert %}}

Open the **Autoscaler Configurator** by clicking the **Configurator** next to the autoscaler service in the [Services tab](https://docs.mendix.com/developerportal/deploy/sap-cloud-platform/) of the Environment Details.

You first need to decide the sort of scaling you want to apply.

For **Scaling**, select one of the following:

> * Schedule-based scaling – scaling based on a schedule, see [Scheduled Scaling](#schedule) section below

Reviewer: You need a 'the' here. You can see "Scheduled Scaling", but you see "the Scheduled Scaling section"
```suggestion
* Schedule-based scaling – scaling based on a schedule, see the [Scheduled Scaling](#schedule) section below
```

> * Dynamic scaling – scaling based on the values of metrics, see [Dynamic Scaling](#dynamic)section below

Reviewer: ```suggestion
* Dynamic scaling – scaling based on the values of metrics, see [Dynamic Scaling](#dynamic) section below
```

Author: done https://github.com/mendix/docs/pull/6673/commits/be785809847f6b0b1d0a54e88bccaceb697d2a72

* Custom – you need to set up both [Dynamic Scaling](#dynamic) and [Scheduled Scaling](#schedule) as described in the sections below

You then need to set up the requested type of scaling. The sections below show a summary of the required information. For more information, see the [Application Autoscaler Service](https://help.sap.com/docs/CSI/b651ed9a533d41339f05b12549d8d706/c5d349a5ea354971839b7922d13c04e4.html?q=application%20autoscaler%20service) documentation on the SAP Help Portal.

When you have added all the rules, you can do one of the following:

* Click **Upload Configuration To Service** to upload your autoscaler configuration directly to the autoscaler service.

> * Click **Download Configuration File** to create a file containing the correct JSON for your autoscaler configuration; you can then upload this file as a configuration of your autoscaler service.

Reviewer: I think this is still **the** configuration. Although you can create multiple configurations, I think you can only upload one configuration to the service.
```suggestion
* Click **Download Configuration File** to create a file containing the correct JSON for your autoscaler configuration; you can then upload this file as the configuration of your autoscaler service.
```

Author: done https://github.com/mendix/docs/pull/6673/commits/be785809847f6b0b1d0a54e88bccaceb697d2a72

### 3.1 Scheduled Scaling{#schedule}

{{% alert color="warning" %}}

> Do not attempt to enter overlapping schedules, as it causes the scaling to fail.

Reviewer: I don't think this comma is needed.

## 1 Introduction

> The application autoscaler service for [SAP Business Technology Platform (SAP BTP)](https://docs.mendix.com/developerportal/deploy/sap-cloud-platform/) enables applications to automatically increase or decrease the number of instances of your app. This service can be based on a schedule, or based on the value of a metric.

Author: Verified with developer that 'application autoscaler' is not an UI text.

Reviewer: Great to remove the bold.
I'm not sure that it is the _service_ which is based on the schedule - it is the automatic change of resources that is based on the schedule.
Perhaps
```suggestion
The application autoscaler service for [SAP Business Technology Platform (SAP BTP)](https://docs.mendix.com/developerportal/deploy/sap-cloud-platform/) enables applications to automatically increase or decrease the number of instances of your app. These changes can be based on a schedule, or based on the value of a metric.
```

Author: I agree. Corrected in https://github.com/mendix/docs/pull/6673/commits/be785809847f6b0b1d0a54e88bccaceb697d2a72
